### PR TITLE
[PublishCodeCoverageResultsV1] Migrated to Node 10

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
**Task name**: PublishCodeCoverageResultsV1

**Description**: Migrated the PublishCodeCoverageResultsV1 task to Node10 execution handler

**Documentation changes required:** No

**Added unit tests:** Yes (migrated legacy tests)

**Attached related issue:** [#356](https://github.com/microsoft/build-task-team/issues/356)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by DaniilShmelev** [comment]
should we also remove the legacy tests?

File: Tasks/PublishCodeCoverageResultsV1/package.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
   "dependencies": {
-    "azure-pipelines-task-lib": "2.8.0",
-    "azure-pipelines-tasks-codecoverage-tools": "1.0.0"
+    "@types/node": "^10.17.0",
+    "@types/mocha": "^5.2.7",
+    "azure-pipelines-task-lib": "3.0.6-preview.1",
```
Comment (anatolybolshakov) [inline Tasks/PublishCodeCoverageResultsV1/package.json:5]: Could it make sense to specify it as '^3.0.6-preview.1'?

Comment (egor-bryzgalov) [reply Tasks/PublishCodeCoverageResultsV1/package.json]: Thanks! Fixed it

**Review by EzzhevNikita — APPROVED** [review]
LGTM!

File: Tasks/PublishCodeCoverageResultsV1/package.json [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {
   "dependencies": {
-    "azure-pipelines-task-lib": "2.8.0",
-    "azure-pipelines-tasks-codecoverage-tools": "1.0.0"
+    "@types/node": "^10.17.0",
+    "@types/mocha": "^5.2.7",
+    "azure-pipelines-task-lib": "^3.0.6-preview.1",
+    "azure-pipelines-tasks-codecoverage-tools": "2.0.0-preview.0"
```
Comment (anatolybolshakov) [inline Tasks/PublishCodeCoverageResultsV1/package.json:6]: Could you please double check that there are no any other changes not related to migration between 1.0.0 and 2.0.0-preview.0 versions?

Comment (egor-bryzgalov) [reply Tasks/PublishCodeCoverageResultsV1/package.json:6]: @anatolybolshakov  `2.0.0-preview.0` is based on the latest stable version of the package. There's also no task-lib versions conflicts.

**Comment by egor-bryzgalov** [comment]
Conducted manual testing based on the [canary pipeline definition](https://buildcanary.visualstudio.com/PipelineCanary/_build?definitionId=4249&_a=summary).
Checked both jacoco and cobertura coverage tools. Covered other inputs related features including `additionalCodeCoverageFiles` option and `reportDirectory` so all tasks' inputs were used.
[Successful build example](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1509&view=results).

**Review by DaniilShmelev — APPROVED** [review]
[Self-hosted windows and centos with no spaces](https://dev.azure.com/v-dshmelev/playground/_build/results?buildId=651&view=results)
[Self-hosted windows and centos with spaces](https://dev.azure.com/v-dshmelev/playground/_build/results?buildId=653&view=results)

LGTM!

**Pull request merged**